### ISSUE TITLE
Login while on logout page displays misleading message

### ISSUE THREAD

**Issue by tuxmaster5000** [jenkinsci/oic-auth-plugin#295]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Paste the output here
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Rock 9.3

### Reproduction steps

1. Log in
2. log out
3. re login

### Expected Results

That after logging out the token on the IDP will be removed and all session data

### Actual Results

The  user will be still logged in.

### Anything else?

Picture after step 2:
![step2](https://github.com/jenkinsci/oic-auth-plugin/assets/837503/30ec2b6c-cfb6-4d20-af32-6d011469239f)

Picture after step 3:
![step3](https://github.com/jenkinsci/oic-auth-plugin/assets/837503/cf41d8f1-564a-4cfb-8864-f968d39e538d)

**Comment by michael-doubez** [comment]
Can you please provide more details ?

- regression or new setup ?
- version of jenkins and plugin ?
- mono instance ?
- oidc provider ?
- which fields do you use in config ?

The logout of the user is handled by Jenkins itself, I ll check what how we can end up in this situation (sessions were changed recently so it may be a side effect)

**Comment by michael-doubez** [comment]
I think I understand. When you login in the third step it redirects you to the logout landing page because it is the page you login from.

It *is* the expected behavior but indeed a bit misleading :D

I ll make a fix in the redirect to redirect to the root url if the redirect url is the logout landing page.

**Comment by michael-doubez** [comment]
Can you confirm that if you go to another page before logging back, you don t have this message ?

**Comment by tuxmaster5000** [comment]
@michael-doubez ,
before I switched to OpenIDC the system was using ldap before for the user authorisation.
Versions:
Jenkins:  2.453
Add-on: 4.239.v325750a_96f3b_ 
What do you mean with mono instance? cluster/stand alone? It is an single installation.
oidc provider: Keycloak 24.0.2
Config:
User name field name -> preferred_username
Groups field name -> realm_access.roles[].name
Post logout redirect URL -> Main company page
Enable the 'escape" config in case OpenID will not work.
The other fields are default.

I click on logout,  then the first picture will shown, and I have the Login button again.
(On the IDP side I see, that the session is still valid.) When I now click on login again, the second picture will be presented.

**Comment by michael-doubez** [comment]
When you login, the plugin redirects you to the page you started from. So when you login from the logout page it displays it but the page logic expects you to be logged out. That s why you have this message.

I ll change the login logic to redirect to root url in this specific case.

**Comment by tuxmaster5000** [comment]
Thanks.
And what is with the undeleted  session on the idp? Or should I open a separate task for this?

**Comment by michael-doubez** [comment]
You mean the token was not revoked ?

**Comment by tuxmaster5000** [comment]
Yes, that is an another name for the same. (Under keycloak this will be called session).
But where can I set this? I don't see an option for that:
![Screenshot 2024-04-16 at 11-31-59 Security Jenkins](https://github.com/jenkinsci/oic-auth-plugin/assets/837503/9e62f6b7-ad2d-41b3-86fa-5dccbf74f3de)

**Comment by michael-doubez** [comment]
It shows up when you click on "Manual configuration" ...
Damned. I'll have to fix that.

Click on manual, activate it and swithc back to "Automatic configuration", it will be kept.

**Comment by tuxmaster5000** [comment]
Switching back will not work, because the settings will be removed:
![Screenshot 2024-04-17 at 10-26-36 Security Jenkins](https://github.com/jenkinsci/oic-auth-plugin/assets/837503/f4e22a3a-92a0-4881-b19c-8b17f922b8b1)

**Comment by tuxmaster5000** [comment]
Now the login works under the UI, but on the IDP not all sessions are removed.
On login, two are created, but only one will be removed:
![Screenshot 2024-04-17 at 10-37-31 Keycloak Administration UI](https://github.com/jenkinsci/oic-auth-plugin/assets/837503/7103031a-da2d-48e5-b132-c6012e971afe)
The regular one will be removed, but the offline will be still present.

**Comment by michael-doubez** [comment]
Ah yes. That's because you requested the refresh token which is not handled at this point.

The reason it is requested is because the wellknown configuration requests all available scopes, including `offline_token` which in your case triggers the generation of the offline token. It didn't know it resulted in a open session but it make sense.

You can fix that by overriding the scopes to specify only what you need: tick the "override Scopes" box in config and remove `offline_token`.

**Comment by tuxmaster5000** [comment]
Hi @michael-doubez  yes this will fix it. 
I think it would be helpful to include this in the documentation. (Because I had users with over 30 offline sessions/token on the SSO server today)
